storio-sqlite/src/test/java/com/pushtorefresh/storio/sqlite/operations/delete/PreparedDeleteByQueryTest.java

```diff
@@ -0,0 +1,161 @@
+package com.pushtorefresh.storio.sqlite.operations.delete;
+
+import com.pushtorefresh.storio.StorIOException;
+import com.pushtorefresh.storio.sqlite.Changes;
+import com.pushtorefresh.storio.sqlite.StorIOSQLite;
+import com.pushtorefresh.storio.sqlite.queries.DeleteQuery;
+
+import org.junit.Test;
+
+import rx.observers.TestSubscriber;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Matchers.same;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+public class PreparedDeleteByQueryTest {
+
+    @Test
+    public void shouldPerformDeletionByQueryBlocking() {
+        final StorIOSQLite storIOSQLite = mock(StorIOSQLite.class);
+        final StorIOSQLite.Internal internal = mock(StorIOSQLite.Internal.class);
+
+        when(storIOSQLite.internal()).thenReturn(internal);
+
+        final DeleteQuery deleteQuery = DeleteQuery.builder()
+                .table("test_table")
+                .where("column1 = ?")
+                .whereArgs(1)
+                .build();
+
+        //noinspection unchecked
+        final DeleteResolver<DeleteQuery> deleteResolver = mock(DeleteResolver.class);
+
+        final DeleteResult expectedDeleteResult = DeleteResult.newInstance(1, deleteQuery.table());
+
+        when(deleteResolver.performDelete(same(storIOSQLite), same(deleteQuery)))
+                .thenReturn(expectedDeleteResult);
+
+        final DeleteResult actualDeleteResult = new PreparedDeleteByQuery.Builder(storIOSQLite, deleteQuery)
+                .withDeleteResolver(deleteResolver)
+                .prepare()
+                .executeAsBlocking();
+
+        assertEquals(expectedDeleteResult, actualDeleteResult);
+
+
+        verify(storIOSQLite).internal();
+        verify(deleteResolver).performDelete(same(storIOSQLite), same(deleteQuery));
```

Comment:
Because `same()` is more strict check than `eq()`, it's like `ref1 == ref2` vs `ref1.equals(ref2)`